### PR TITLE
Refs #22164 - use cheaper FactoryBot methods

### PR DESCRIPTION
**Pull request by mmoll**
This is mainly for discussion.

Especially the change from create to build reduces database access.

In my measurements these changes lead to about 5% faster execution.

### CODE REVIEW

**Comment by mmoll** [comment]
https://robots.thoughtbot.com/use-factory-girls-build-stubbed-for-a-faster-test

**Review by iNecas — APPROVED** [review]

**Comment by mmoll** [comment]
Test failure unrelated I'd say. Just to clarify: The ~%5 speed up was for `test/models/shared/taxonomies_base_test.rb` and `test/controllers/hosts_controller_test.rb`.

It's too tedious for me to further replace the methods (the approach is basically to replace the methods, run the tests and then revert back until everuthing is green again), so I propose to merge this (there might be some fallout for plugins) and advise developers at least for new tests to use the expensive methods only when needed.

File: test/active_support_test_case_helper.rb [hunk 1]
```diff
@@ -131,7 +131,7 @@ def setup_user(operation, type = "", search = nil, user = :one)
     @one = users(user)
     as_admin do
       permission = Permission.find_by_name("#{operation}_#{type}") ||
```
Comment (ekohl) [inline test/active_support_test_case_helper.rb:133]: If the permission doesn't have to be in the database, is finding it in the database even needed then?

**Review by lzap — APPROVED** [review]
Nice.

**Review by dLobatog — APPROVED** [review]
Thanks @mmoll , huge manual effort here - hopefully other people will notice, I try to emphasize this on PRs, bit by bit it eats up valuable testing time.

**Comment by dLobatog** [comment]
This passed tests, notice just on 2.3 mysql it was unable to install gems

```
+ return 0
+ bundle install --without=development -j 5
FATAL: command execution failed
java.io.EOFException
	at java.io.ObjectInputStream$PeekInputStream.readFully(ObjectInputStream.java:2680)
	at java.io.ObjectInputStream$BlockDataInputStream.readShort(ObjectInputStream.java:3155)
	at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:864)
	at java.io.ObjectInputStream.<init>(ObjectInputStream.java:360)
	at hudson.remoting.ObjectInputStreamEx.<init>(ObjectInputStreamEx.java:48)
	at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:35)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(Sync
```

**Comment by lzap** [comment]
image

Bravo!

**Comment by ekohl** [comment]
@lzap what is that counting exactly?

**Comment by lzap** [comment]
Haha that's execution time, these are seconds I guess (110 minutes). Taken from our Jenkins when you drill down to jobs.

**Comment by mmoll** [comment]
I think I found that graph in Jenkins and guess this is rather due to the API v1 removal.

**Comment by ekohl** [comment]
With count I'd expect an actual unit test count rather than seconds.

**Comment by mmoll** [comment]
> With count I'd expect an actual unit test count rather than seconds.

It is actually the test count.

**Pull request merged**